### PR TITLE
Docs: Add warning about sharing a syntax highlighter to `TextEdit`

### LINKED ISSUE #81419: Multiple CodeEdits make infinite loop on request_code_completion()
### Godot version

4.1.1

### System information

Windows 11

### Issue description

If you have more than one CodeEdit, all of the other CodeEdits will go off of the last CodeEdit created. For example, if you have two CodeEdits and you use ` as the start of an add_color_region it will update that exact column and forward for all code edits. It also works for keywords as well. So for example if the first 5 letters on the last CodeEdit are red, then the first 5 letters will be red for all.
Screenshot 2023-09-07 084340. If you add a print statement to the _on_code_completion_requested() it will print non-stop once you have more than one Code Edit.

https://github.com/godotengine/godot/assets/29180106/ea95fa5d-c528-48bc-8842-004db4149f1f

### Steps to reproduce

Have two or more CodeEdits with syntax highlighter and code completion enabled. 

### Minimal reproduction project

[Bug Report.zip](https://github.com/godotengine/godot/files/12549881/Bug.Report.zip)

### PR DESCRIPTION
**Pull request by HolonProduction**
Closes #81419

### CODE REVIEW

**Comment by akien-mga** [comment]
Makes sense to document, but that sounds a bit counterintuitive to me. What's the point of having SyntaxHighlighter be a Resource, if it can't be shared?

**Review by akien-mga — APPROVED** [review]

**Comment by Mickeon** [comment]
> What's the point of having SyntaxHighlighter be a Resource, if it can't be shared?

I believe it's the same case as **TileSetSource** & co. which is, indeed, very annoying in practice. Still worth documenting in the hopes that this behavior is fixed someday.

File: doc/classes/TextEdit.xml [hunk 1]
```diff
@@ -1362,6 +1362,7 @@
 		</member>
 		<member name="syntax_highlighter" type="SyntaxHighlighter" setter="set_syntax_highlighter" getter="get_syntax_highlighter">
 			Sets the [SyntaxHighlighter] to use.
```
Comment (Mickeon) [inline doc/classes/TextEdit.xml:1364]: While you're here:
This is worded like a method, a setter, for some reason. And repeating the property's type makes it a bit annoying to localise.
```suggestion
			The syntax highlighter to use.
```

**Review by Mickeon — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**